compiler/src/dmd/escape.d

```diff
@@ -2171,6 +2144,29 @@ void escapeByRef(Expression e, EscapeByResults* er, bool live = false, bool retR
     }
 }
 
+/**
+ * Iterate over the arguments and corresponding parameters,
+ * ignoring the hidden `_arguments[]` argument of a DstyleVariadic.
+ *
+ * Params:
+ *   tf = function type
+ *   args = arguments
+ *   dg = delegate to call for each argument
+ */
+private void foreachEscapableArgument(TypeFunction tf, Expressions* args, void delegate(Parameter, Expression) dg) {
+    if (!args)
+        return;
```

Comment:
Move away from passing args by pointer to by ref. This means the above test will have to be done before the function is called.